src/SourceLink.Tools/SourceLinkMap.cs

```diff
@@ -209,20 +228,24 @@ private static bool TryParseEntry(string key, string value, out Entry entry)
                 {
                     if (path.StartsWith(file.Path, StringComparison.OrdinalIgnoreCase))
                     {
-                        var escapedPath = string.Join("/", path.Substring(file.Path.Length).Split(new[] { '/', '\\' }).Select(Uri.EscapeDataString));
+                        var relativePath = path.Substring(file.Path.Length).Replace('\\', '/');
+                        var escapedPath = string.Join("/", relativePath.Split('/').Select(Uri.EscapeDataString));
                         uri = mappedUri.Prefix + escapedPath + mappedUri.Suffix;
+                        relativeFilePath = relativePath;
                         return true;
                     }
                 }
                 else if (string.Equals(path, file.Path, StringComparison.OrdinalIgnoreCase))
                 {
                     Debug.Assert(mappedUri.Suffix.Length == 0);
+                    relativeFilePath = Path.GetFileName(file.Path);
```

Comment:
@tmat Mac tests don't like this because the test input is windows style paths and I'm assuming Path.GetFileName(...) looks for forward slashes on that OS.

should this just be something like
```cs
                    int lastSeparatorIndex = file.Path.LastIndexOfAny(new[] { '\\', '/' });
                    relativeFilePath = lastSeparatorIndex > 0 ? file.Path.Substring(lastSeparatorIndex + 1) : file.Path;
```

Reply:
Yes, can't use GetFileName.

Reply:
Can we set `relativeFilePath = null` here? Since there is no relative path... the caller can use the file name, if they want to.